model/godleyTableWindow.cc

```diff
@@ -995,7 +1001,7 @@ namespace {
                 {
                   auto savedItem=minsky().canvas.item;
                   minsky().canvas.item=sv;
-                  minsky().canvas.renameAllInstances(godleyIcon->table.cell(selectedRow,selectedCol));
+                  minsky().canvas.renameAllInstances(utf_to_utf<char>(godleyIcon->table.cell(selectedRow,selectedCol)));
```

Comment:
Could you enlighten me as to what utf_to_utf<char> does on an 8 bit string? Nominally, it should just be the identity transformation, but does it somehow clean up non UTF-8 characters also?

Reply:
Ah - it took me a while to find this:

This function may use an explicit encoding name like "Latin1" or "ISO-8859-8", or use std::locale as a parameter to fetch this information from it. It also receives a policy parameter that tells it how to behave if the conversion can't be performed (i.e. an illegal or unsupported character is found). By default this function skips all illegal characters and tries to do the best it can, however, it is possible ask it to throw a conversion_error exception by passing the stop flag to it:

std::wstring s=to_utf<wchar_t>("\xFF\xFF","UTF-8",stop); 
// Throws because this string is illegal in UTF-8

Reply:
So the answer is that it simply ignores invalid utf-8 characters. Neat!

Reply:
I'm glad this one has been sorted out :)